core/src/utils/sync/racy_lock.rs

```diff
@@ -0,0 +1,148 @@
+use alloc::boxed::Box;
+use core::{
+    ops::Deref,
+    ptr,
+    sync::atomic::{AtomicPtr, Ordering},
+};
+
+/// Thread-safe, non-blocking, "first one wins" flavor of `once_cell::sync::OnceCell`
+/// with the same interface as `std::sync::LazyLock`.
+///
+/// The underlying implementation is based on `once_cell::sync::race::OnceBox` which relies on
+/// `core::atomic::AtomicPtr` to ensure that the data race results in a single successful
+/// write to the relevant pointer, namely the first write.
+/// See https://github.com/matklad/once_cell/blob/v1.19.0/src/race.rs#L294.
+///
+/// Performs lazy evaluation and can be used for statics.
+pub struct RacyLock<T, F = fn() -> T>
```

Comment:
We should implement `Debug` and `Default` to match the implemented traits of `LazyLock`

Reply:
Updated